crates/flux-syntax/src/surface.rs

```diff
@@ -110,6 +109,27 @@ pub struct TyAlias {
     pub span: Span,
 }
 
+#[derive(Debug)]
+pub struct DetachedSpecs {
+    pub items: Vec<DetachedItem>,
+}
+
+#[derive(Debug)]
+pub enum DetachedItem {
```

Comment:
I'm hoping we can reuse this definition for normal specs so we can make the code in name resolution nicer, so I prefer to call this simply `Item` (and rename the current `Item` to `FluxItem`). But we can always rename this later.

Right now we have separate functions for different type of items, but they all behave uniformly, e.g.
https://github.com/flux-rs/flux/blob/8ff117f14ed85215256bb4f60961abc06ccbbe49/crates/flux-desugar/src/resolver.rs#L314)
https://github.com/flux-rs/flux/blob/8ff117f14ed85215256bb4f60961abc06ccbbe49/crates/flux-desugar/src/resolver.rs#L322
If we had a node for items we could just visit the ast.

Reply:
done!